pkg/controllers/vdb/init_db.go

```diff
@@ -540,3 +646,30 @@ func (g *GenericDatabaseInitializer) hasCompatibleVersionForKerberos() ctrl.Resu
 		vinf.VdbVer, DefaultKerberosSupportedVersion)
 	return ctrl.Result{Requeue: true}
 }
+
+// hasCompatibleVersionForServerSideEncryption checks whether it has the required engine fix
+// to run with S3 server-side encryption.  If it doesn't the ctrl.Result will have the
+// requeue bool set.
+func (g *GenericDatabaseInitializer) hasCompatibleVersionForServerSideEncryption() ctrl.Result {
+	vinf, ok := g.Vdb.MakeVersionInfo()
+	const DefaultSseSupportedVersion = "v12.0.1"
+	if !ok || ok && vinf.IsEqualOrNewer(DefaultSseSupportedVersion) {
+		return ctrl.Result{}
+	}
+	g.VRec.Eventf(g.Vdb, corev1.EventTypeWarning, events.UnsupportedVerticaVersion,
+		"The engine (%s) doesn't have the required change for server-side encryption. "+
+			"You must be on version %s or greater",
+		vinf.VdbVer, DefaultSseSupportedVersion)
+	return ctrl.Result{Requeue: true}
+}
+
+func genCommunalParmsMap(content string) map[string]string {
+	parmNames := map[string]string{}
+	for _, row := range strings.Split(content, "\n") {
+		if row != "" {
+			s := strings.Split(row, " = ")
+			parmNames[strings.ToLower(s[0])] = s[1]
```

Comment:
For example `awsauth` and `AWSauth` are the same. The to lowercase is to avoid this case. I think we should also check additionalConfig keys for this kind of duplication:
```
additionalConfig:
    awsauth: "ffff:fff"
    AWSAuth: "aaaa:aaaa"

```
We should make sure only is passed into `auth_conf`. If both are passed, the 1st one in the file will be applied.

Reply:
Okay, please add a comment to explain why this is done.